server/Dockerfile

```diff
@@ -0,0 +1,15 @@
+FROM node:16.16.0-alpine3.15 as common-build-stage
+
+# Create app directory
+RUN mkdir -p /usr/src/app
+WORKDIR /usr/src/app
+
+# Install app dependencies
+COPY package.json /usr/src/app/
+RUN npm install
+
+# Bundle app source
+COPY . /usr/src/app
```

Comment:
Is cool that you copy first the pacage.json for docker to cache the install, but then you can not copy the whole project as you do the second time. 

That way you will override the node_modules and package-lock with the ones you have in local. They could be outdated.

Reply:
I think I've undrestood what you mean here and Ifixed that in the code but check if I did it correctly please